airrohr-firmware/airrohr-firmware.ino

```diff
@@ -3066,6 +3131,74 @@ static void fetchSensorPMS(String& s) {
 	debug_outln_verbose(FPSTR(DBG_TXT_END_READING), FPSTR(SENSORS_PMSx003));
 }
 
+
+/*****************************************************************
+ * parse  Plantronic PM  sensor values  for DEBUG                *
+ *****************************************************************/
+void parsePMSPayloadForDebug(String &pms_data){
+	int start_PMS_P0 = pms_data.indexOf('{');
+	int end_PMS_P0 = pms_data.indexOf('}',start_PMS_P0);
+	String PMS_P0_value = pms_data.substring(start_PMS_P0,end_PMS_P0);
+
+	DynamicJsonDocument P0_doc(1024);
+	deserializeJson(P0_doc,PMS_P0_value);
+  	JsonObject P0_obj = P0_doc.as<JsonObject>();
+	String P0_value = P0_obj["value"];
+
+	int start_PMS_P1 = pms_data.indexOf('{',end_PMS_P0);
+	int end_PMS_P1 = pms_data.indexOf('}',start_PMS_P1);
+	String PMS_P1_value = pms_data.substring(start_PMS_P1,end_PMS_P1);
+	
+	DynamicJsonDocument P1_doc(1024);
+	deserializeJson(P1_doc,PMS_P1_value);
+  	JsonObject P1_obj = P1_doc.as<JsonObject>();
+	String P1_value = P1_obj["value"];
+
+	int start_PMS_P2 = pms_data.indexOf('{',end_PMS_P1);
+	int end_PMS_P2 = pms_data.indexOf('}',start_PMS_P2);
+	String PMS_P2_value = pms_data.substring(start_PMS_P2,end_PMS_P2);
+	
+	DynamicJsonDocument P2_doc(1024);
+	deserializeJson(P2_doc,PMS_P2_value);
+  	JsonObject P2_obj = P2_doc.as<JsonObject>();
+	String P2_value = P2_obj["value"];
+
+	last_value_PMS_P0 = P0_value.toFloat();
+	last_value_PMS_P1 = P1_value.toFloat();
+	last_value_PMS_P2 = P2_value.toFloat();
+
+	debug_outln_info(FPSTR("PM1 (sec.): "), String(last_value_PMS_P0));
+	debug_outln_info(FPSTR("PM2.5 (sec.): "), String(last_value_PMS_P2));
+	debug_outln_info(FPSTR("PM10 (sec.) : "), String(last_value_PMS_P1));
+
+
+}
+
+/*****************************************************************
+ * read Plantronic PM sensor sensor values  from ATMEGA                     *
+ *****************************************************************/
+String fetchSensorPMSFromAtmega(){
+	debug_outln_verbose(FPSTR(DBG_TXT_START_READING), FPSTR(SENSORS_PMSx003));
+	RESERVE_STRING(s,SMALL_STR);
+
+	if(send_now){
+		atmega328p.println("fetchSensorPMS");
+		delay(WARMUPTIME_SDS_MS + READINGTIME_SDS_MS + 2000);
+		while(atmega328p.available() > 0){
+			String pms_data = atmega328p.readString();
```

Comment:
@MCLEANS do you have an example of what string is returned here?

Reply:
@chegejames 
![pms_payload](https://user-images.githubusercontent.com/33148990/87017546-c7f46200-c1d8-11ea-9abd-ab610090e472.PNG)